### ISSUE TITLE
[V4] Error on useModal

### ISSUE THREAD

**Issue by v1r0x** [vue-final/vue-final-modal#277]
I was looking for a way to update props/attrs of a dynamic modal from outside the modal. Luckily I found an existing issue stating that this will be part of next major release.
Thus, I migrated from v3 to v4 (RC) and followed the migration guide to switch from `$vfm.show()` to `useModal()` for dynamic modals.
But calling `useModal()` leads to an error in my browser's console (see below). Not sure if I did something wrong or if it is really a bug.

### Version

vue-final-modal: v4.0.0-rc.0
vue: 3.2.44
nuxt: Not installed

### OS

Linux

### Steps to reproduce
1. Install `vue-final-modal`
    ```vue
    import { createVfm } from 'vue-final-modal';
    const vfm = createVfm();
    app.use(vfm);
    ```
2. Call `useModal()` to create a modal
    ```vue
    import { useModal } from 'vue-final-modal';

    const {
        open,
        close,
        patchOptions,
    } = useModal({
        component: AddUser,
        attrs: {},
    });
    ```
3. Open page and try to open modal

### What is Expected?
Modal should open normally

### What is actually happening?
Modal does not open, error is thrown instead in the browser console (firefox in my case)
```
Uncaught TypeError: ae() is undefined
    Lt                        index.es.js:12
    showModal         modal.js:279
```

**Comment by v1r0x** [comment]
I finally got it working (at least somehow).
With v3 I could call `$vfm.show()` from any js file, but `useModal()` only works in `setup()` from a custom component.
In v3 I did it like so:
```javascript
# modal.js
export function showModal() {
   $vfm.show({
        component: MyComponent,
    });
}
```
This no longer works. Is there any way to get that behaviour back or do I have to explicitly use it a component's `setup()` method?

**Comment by hunterliu1003** [comment]
I think it will look like this

```js
// useMyModal.js
export function useMyComponent() {
  return useModal({
        component: MyComponent,
    })
}
```

```vue
<script setup>
const { open } = useMyModal()
</script>

<template>
  <button @click="open">open modal</button>
</template>
```

I understand that is annoying to always use `useModal()` first, in some use cases we don't know when the modal will be opened. I am still trying to find a way to create a dynamic modal from outside of setup script.

**Comment by v1r0x** [comment]
Hey @hunterliu1003 
Thanks for your quick reply!
Alright, as long as it works I'm fine with it :wink: 
I'll give it a try next week and see how it works.

**Comment by hunterliu1003** [comment]
Hey @v1r0x 

How about providing second argument for useModal. The second argument should be a `vfm` instance create by `createVfm()`. If passing a vfm as second argument to `useModal()`, it will use vfm instance instead of `injectedVfm`.

```js
// ./plugins/vue-final-modal.js
import { createVfm } from 'vue-final-modal'

export const vfm = createVfm()
```

```js
// main.js
import { vfm } from '@/plugins/vue-final-modal'
app.use(vfm)
```

```js
// Anywhere outside of script setup
import { vfm } from '@/plugins/vue-final-modal'
const modal = useModal({
  component: MyComp,
  ...
}, vfm)

modal.open()
```

But you have to manually execute `modal.destroy()` whenever you don't need it anymore. Otherwise, the length of the vfm.dynamicModals will getting larger and that might cause memory leak (see #206).

I think in your case, you can execute `modal.destroy()` on `onClosed` hook:
```js
// Anywhere outside of script setup
import { vfm } from '@/plugins/vue-final-modal'
const modal = useModal({
  component: MyComp,
  attrs: {
    onClosed() {
      modal.destroy()
    }
  }
}, vfm)

modal.open()
```

**Comment by v1r0x** [comment]
Hey @hunterliu1003 

passing the `vfm` instance as option param instead of another optional param is also fine :)
If this is possible it would help me a lot!

Also calling `modal.destroy()` in `onClosed()` is fine for me. Would this make `modal.close()` obsolete in my case?

**Comment by hunterliu1003** [comment]
Hey @v1r0x

> Also calling `modal.destroy()` in `onClosed()` is fine for me. Would this make `modal.close()` obsolete in my case?

1. `modal.destroy()` will delete the modal instance inside of `vfm.dynamicModals`. So if you have an opened modal, it will be closed immediate without calling `modal.close()`.
2. Calling `modal.close()` will not delete the modal instance inside of `vfm.dynamicModals`. So if you keep calling the `showModal()`, the length of `vfm.dynamicModals` will keep growing until you have memory leak (must be careful in Nuxt). This is the reason why you have to manually call `modal.destroy()` when you don't need this dynamic modal instance anymore (usually on `onClosed` hook I think).
3. Without calling `modal.destroy()` correctly, the memory leak might occur on server side (Nuxt 3) and on client side (Vue 3 SPA).

> Edit: I saw you already pushed that to `v4` branch. Awesome! 🎉 Is there a way to test this? I tried adding that branch using npm, but that doesn't work.

I just released [vue-final-modal@4.0.0-rc.1](https://www.npmjs.com/package/vue-final-modal/v/4.0.0-rc.1), so you can test it. I'll finished the documentation on this weekend (with no guarantee).

**Comment by v1r0x** [comment]
Thank you for your fast response and new release! :heart: 
I tried to install the new release through npm, but ran into several problems.
First, I didn't have `vue-tsc` installed and had to install it manually. After that I got that error:
```bash
$ npm install
npm ERR! code 1
npm ERR! path my_project/node_modules/vue-final-modal
npm ERR! command failed
npm ERR! command sh -c vue-tsc --noEmit && vite build
npm ERR! Version 4.9.3
npm ERR! tsc: The TypeScript Compiler - Version 4.9.3
npm ERR! 
npm ERR! COMMON COMMANDS
...
```

And then it simply prints the available commands for tsc. I also tried to run `tsc` by hand. This also prints the available commands :confused: 
I'm on node v16 and npm v9 and installed the latest `vue-tsc` and `tsc`. Any ideas what I could have done wrong?
